lib/modules/versioning/types.ts

```diff
@@ -10,26 +10,88 @@ export interface NewValueConfig {
 }
 export interface VersioningApi {
   // validation
+
+  /**
+   * Check whether the `version` is compatible with the `current` version range
+   * constraint.
+   */
   isCompatible(version: string, current?: string): boolean;
+
+  /**
+   * Check whether the `version` constraint is pinned, i.e. it only allows a
+   * single specific version.
+   */
   isSingleVersion(version: string): boolean;
+
+  /**
+   * Check whether the `version` is considered to be "stable".
+   *
+   * Example: in SemVer the version must not have a pre-release marker.
+   */
   isStable(version: string): boolean;
+
+  /**
+   * Check whether the `input` is a valid version range constraint.
+   */
   isValid(input: string): boolean;
+
+  /**
+   * Check whether the `input` is a valid version string.
+   */
   isVersion(input: string | undefined | null): boolean;
 
   // digestion of version
+
   getMajor(version: string | SemVer): null | number;
   getMinor(version: string | SemVer): null | number;
   getPatch(version: string | SemVer): null | number;
 
   // comparison
+
+  /**
+   * Check whether `version` and `other` are logically equivalent, even if
+   * they're not the exact same string.
```

Comment:
eg semver v2 build metadata should be ignored when comparing

Reply:
```suggestion
   * they're not the exact same string.
   *
   * For example, with Semver the build metadata should be ignored when comparing.
```

like this?